src/app/games/hive/HiveMove.ts

```diff
@@ -97,59 +98,41 @@ export class HiveMovePass extends Move {
     }
 }
 
-export type HiveMove = HiveMoveDrop | HiveMoveCoordToCoord | HiveMoveSpider | HiveMovePass;
+export type HiveMove = HiveDropMove | HiveMoveCoordToCoord | HiveMoveSpider | HiveMovePass;
 
 // eslint-disable-next-line @typescript-eslint/no-redeclare
 export namespace HiveMove {
 
     export const PASS: HiveMove = new HiveMovePass();
 
     export function drop(piece: HivePiece, coord: Coord): HiveMove {
-        return HiveMoveDrop.from(piece, coord);
+        return HiveDropMove.of(piece, coord);
     }
     export function move(start: Coord, end: Coord): MGPFallible<HiveMove> {
         return HiveMoveCoordToCoord.from(start, end);
     }
     export function spiderMove(coords: [Coord, Coord, Coord, Coord]): HiveMove {
-        return HiveMoveSpider.fromCoords(coords);
-    }
-    export const encoder: MoveEncoder<HiveMove> = new class extends MoveEncoder<HiveMove> {
-        public encodeMove(value: HiveMove): JSONValueWithoutArray {
-            if (value instanceof HiveMoveDrop) {
-                return {
-                    moveType: 'Drop',
-                    encoded: HiveMoveDrop.encoder.encode(value),
-                };
-            } else if (value instanceof HiveMoveSpider) {
-                return {
-                    moveType: 'Spider',
-                    encoded: HiveMoveSpider.encoder.encode(value),
-                };
-            } else if (value instanceof HiveMoveCoordToCoord) {
-                return {
-                    moveType: 'CoordToCoord',
-                    encoded: HiveMoveCoordToCoord.encoder.encode(value),
-                };
-            } else {
-                return {
-                    moveType: 'Pass',
-                };
-            }
-        }
-        public decodeMove(encoded: JSONValueWithoutArray): HiveMove {
-            // eslint-disable-next-line dot-notation
-            const moveType: string = Utils.getNonNullable(encoded)['moveType'];
-            // eslint-disable-next-line dot-notation
-            const content: JSONValue = Utils.getNonNullable(encoded)['encoded'] as JSONValue;
-            if (moveType === 'Drop') {
-                return HiveMoveDrop.encoder.decode(content);
-            } else if (moveType === 'Spider') {
-                return HiveMoveSpider.encoder.decode(content);
-            } else if (moveType === 'CoordToCoord') {
-                return HiveMoveCoordToCoord.encoder.decode(content);
-            } else {
-                return HiveMove.PASS;
-            }
-        }
-    };
+        return HiveMoveSpider.ofCoords(coords);
+    }
+    export function isInstanceOfHiveDropMove(value: HiveMove): boolean {
```

Comment:
pas nécessaire de les export peut-être ?

Reply:
Si j'enlève l'export j'ai du "Move function declaration to program root.eslint[no-inner-declarations](https://eslint.org/docs/rules/no-inner-declarations)"
Et j'ai la flemme de chercher :p

Reply:
ah, elle est donc à déplacer hors du namespace, simplement !

Reply:
Voilà, dit moi c'qué tin pinse !
PS: un deuxième bipède à eu un échange de "VRÉMIN !?" avec moi ! Plusieurs fois !!!!!